### ISSUE TITLE
refactor

### ISSUE THREAD

**Issue by Shinyzenith** [Shinyzenith/zigshot#1]
1. Once a working demo is done, clean up main struct and remove fields that are no longer needed.
1. And split utils, listeners, and app into three separate files.

**Comment by Shinyzenith** [comment]
finish check_globals

**Comment by Shinyzenith** [comment]
attempt to reduce bitcast and truncates
